### DISCUSSION TITLE
Token refresh API only works when Email Auth is enabled

### DISCUSSION THREAD

**Discussion by davidmuzi** [supabase/supabase#3874] [Questions] [answered]
Im currently using Sign in with Apple, and am able to successfully make user authenticated calls using the returned JWT. However, when I attempt to refresh the token, I get a 400 response.

```
curl -X "POST" "https://avql....supabase.co/auth/v1/token?grant_type=refresh_token" \
     -H 'apikey: eyJhbGci...' \
     -H 'Content-Type: application/json; charset=utf-8' \
     -d $'{"refresh_token": "u14G..."}'
```
Response:
`{"code":400,"msg":"Email logins are disabled"}` 

If I enable Email Auth in settings, then it works, but Im only interested in using Apple login.

**Comment by awalias** [comment, marked as answer]
hey @davidmuzi thanks for this I'm going to create a linked issue for this in our auth backend repo

**Reply by davidmuzi** [reply]
Linked issue: https://github.com/supabase/gotrue/issues/276

**Reply by J0** [reply]
Hey all, the linked issue has been resolved  and token refresh should work independent of whether email auth is enabled. Thanks for your patience!
